frontend/app/src/components/home/PrizeContent.svelte

```diff
@@ -33,19 +46,31 @@
     $: claimedByYou = content.winners.includes($user.userId);
     $: finished = $now500 >= Number(content.endDate);
     $: allClaimed = content.prizesRemaining <= 0;
-    $: disabled = finished || claimedByYou || allClaimed;
+    $: disabled = finished || claimedByYou || allClaimed || !userEligible;
     $: timeRemaining = finished
         ? $_("prizes.finished")
         : client.formatTimeRemaining($now500, Number(content.endDate));
 
+    $: diamondStatus = (
+        content.lifetimeDiamondOnly ? "lifetime" : content.diamondOnly ? "active" : "inactive"
+    ) as DiamondMembershipStatus["kind"];
+
+    $: restrictedPrize =
+        content.diamondOnly ||
+        content.lifetimeDiamondOnly ||
+        content.uniquePersonOnly ||
+        content.streakOnly > 0;
+
+    $: userEligible =
+        (!content.diamondOnly || $isDiamond) &&
+        (!content.lifetimeDiamondOnly || $isLifetimeDiamond) &&
+        (!content.uniquePersonOnly || $user.isUniquePerson) &&
+        content.streakOnly <= client.getStreak($user.userId);
```

Comment:
You might be better off using `$chitStateStore.streak` here as what you have will not be reactive. Not _particularly_ important in this case, but it _could_ give you the wrong result.